clang/test/CodeGenSYCL/offload-fp32-div-sqrt.cpp

```diff
@@ -11,282 +11,289 @@
-// RUN: | FileCheck --check-prefix ROUNDED-SQRT-FAST %s
+// RUN: | FileCheck --check-prefixes=ROUNDED-SQRT-FAST,NFPA-FAST %s
 
 // RUN: %clang_cc1 %{common_opts_spir64} -ffast-math -fno-offload-fp32-prec-div \
 // RUN: -fno-offload-fp32-prec-sqrt %s -o - \
-// RUN: | FileCheck --check-prefix ROUNDED-DIV-FAST %s
+// RUN: | FileCheck --check-prefixes=ROUNDED-DIV-FAST,NFPA-FAST %s
 
 // RUN: %clang_cc1 %{common_opts_spir64} -ffast-math -foffload-fp32-prec-div \
-// RUN: %s -o - | FileCheck --check-prefix PREC-FAST %s
+// RUN: %s -o - | FileCheck --check-prefixes=PREC-FAST,NFPA-FAST %s
 
 // RUN: %clang_cc1 %{common_opts_spir64} -ffast-math \
 // RUN: -fno-offload-fp32-prec-div -foffload-fp32-prec-sqrt %s -o - \
-// RUN: | FileCheck --check-prefix PREC-SQRT-FAST %s
+// RUN: | FileCheck --check-prefixes=PREC-SQRT-FAST,NFPA-FAST %s
 
 // RUN: %clang_cc1 %{common_opts_spir64} -ffast-math \
 // RUN: -fno-offload-fp32-prec-sqrt -foffload-fp32-prec-div \
-// RUN: %s -o - | FileCheck --check-prefix ROUNDED-SQRT-PREC-DIV %s
+// RUN: %s -o - | FileCheck --check-prefixes=ROUNDED-SQRT-PREC-DIV,NFPA-FAST %s
 
 // RUN: %clang_cc1 %{common_opts_spir64} -ffast-math \
 // RUN: -fno-offload-fp32-prec-div -foffload-fp32-prec-sqrt \
-// RUN: %s -o - | FileCheck --check-prefix ROUNDED-DIV-PREC-SQRT %s
+// RUN: %s -o - | FileCheck --check-prefixes=ROUNDED-DIV-PREC-SQRT,NFPA-FAST %s
 
 // RUN: %clang_cc1 %{common_opts_spir64} -ffast-math \
-// RUN: -fno-offload-fp32-prec-div -fno-offload-fp32-prec-sqrt \
-// RUN: %s -o - | FileCheck --check-prefix ROUNDED-DIV-ROUNDED-SQRT-FAST %s
+// RUN: -fno-offload-fp32-prec-div -fno-offload-fp32-prec-sqrt %s -o - \
+// RUN: | FileCheck --check-prefixes=ROUNDED-DIV-ROUNDED-SQRT-FAST,NFPA-FAST %s
 
 // RUN: %clang_cc1 %{common_opts_spir64} -fno-offload-fp32-prec-div \
 // RUN: -ffp-builtin-accuracy=high %s -o - \
-// RUN: | FileCheck --check-prefix LOW-PREC-DIV %s
+// RUN: | FileCheck --check-prefixes=LOW-PREC-DIV,NFPA %s
 
 // RUN: %clang_cc1 %{common_opts_spir64} -ffp-builtin-accuracy=high:div  \
 // RUN: -fno-offload-fp32-prec-div %s -o - \
-// RUN: | FileCheck --check-prefix ROUNDED-DIV %s
+// RUN: | FileCheck --check-prefixes=ROUNDED-DIV,NFPA %s
 
 // RUN: %clang_cc1 %{common_opts_spir64} -fno-offload-fp32-prec-sqrt \
 // RUN: -ffp-builtin-accuracy=high %s -o - \
-// RUN: | FileCheck --check-prefix LOW-PREC-SQRT %s
+// RUN: | FileCheck --check-prefixes=LOW-PREC-SQRT,NFPA %s
 
 // RUN: %clang_cc1 %{common_opts_spir64} -fno-offload-fp32-prec-div \
 // RUN: -ffp-builtin-accuracy=high:sin %s -o - \
-// RUN: | FileCheck --check-prefix ROUNDED-DIV %s
+// RUN: | FileCheck --check-prefixes=ROUNDED-DIV,NFPA %s
 
 // RUN: %clang_cc1 %{common_opts_spir64} -fno-offload-fp32-prec-sqrt \
 // RUN: -ffp-builtin-accuracy=high:sin %s -o - \
-// RUN: | FileCheck --check-prefix ROUNDED-SQRT %s
+// RUN: | FileCheck --check-prefixes=ROUNDED-SQRT,NFPA %s
 
 #include "sycl.hpp"
 
+namespace sycl {
+  class half {};
```

Comment:
IMHO it's important, that we properly define sycl::half class aka to have clang's half storage. So / operator mustn't be extern. Otherwise the code here doesn't test the issue - I'm pretty sure, that _ZdvN4sycl4halfES0_ would never become llvm.fpbuiltin intrinsic anyway.

Reply:
I have added a minimalist `class half`.